azure-functions/acbs-function/src/functions/acbs-amend-facility.js

```diff
@@ -37,100 +37,120 @@ df.app.orchestration('acbs-amend-facility', function* amendFacility(context) {
-           */
-
-          // 1. SOF: Facility Loan Record (FLR)
-          // `Bond` facility only
-          if (facilitySnapshot.type === FACILITY.FACILITY_TYPE.BOND) {
-            facilityLoanRecord = context.df.callSubOrchestrator('acbs-amend-facility-loan-record', {
-              facilityId,
-              facility,
-              amendments,
-              fmr,
-            });
-
-            yield context.df.Task.all([facilityLoanRecord]);
-          } else {
-            facilityLoanRecord = {
-              result: `Facility type ${facilitySnapshot.type} will not be amended.`,
-            };
-          }
-
-          // 2. SOF: Facility Master Record (FMR)
-          const facilityMasterRecord = context.df.callSubOrchestrator('acbs-amend-facility-master-record', {
-            deal,
-            facilityId,
-            fmr,
-            etag,
-            amendments,
-          });
-
-          yield context.df.Task.all([facilityMasterRecord]);
-
-          return {
-            facilityId,
-            facilityLoanRecord: facilityLoanRecord.result,
-            facilityMasterRecord: facilityMasterRecord.result,
-          };
-        }
-        throw new Error('ACBS facility amendment error : Unable to retrieve FMR');
-      }
+
+    if (!amendment) {
+      throw new Error('No amendment provided');
+    }
+
+    const { facilityId, amount, coverEndDate } = amendment;
+
+    // UKEF Facility ID exists in the payload
+    const hasFacilityId = Boolean(facilityId);
+    // At least one of the amendment exists in the payload
+    const hasAmendment = Boolean(amount) || Boolean(coverEndDate);
+    // Facility object existence check
+    const hasFacility = amendment.facility;
+    // Deal properties existence check
+    const hasDeal = amendment.deal && amendment.deal.dealSnapshot;
+
+    // Payload verification
+    if (!hasFacilityId || !hasAmendment || !hasFacility || !hasDeal) {
+      throw new Error('Invalid argument set provided');
+    }
+
+    const { facility, deal } = amendment;
+    const { facilitySnapshot } = facility;
+    let facilityLoanRecord;
+    let facilityCovenantRecord;
+
+    if (facilityId.includes(DEAL.UKEF_ID.PENDING) || facilityId.includes(DEAL.UKEF_ID.TEST)) {
+      throw new Error(`Invalid facility ID ${facilityId}`);
+    }
+
+    // 1. DAF : get-facility-master: Retrieve ACBS `Facility Master Record` with eTag
+    const { acbsFacility: fmr, etag } = yield context.df.callActivityWithRetry('get-facility-master', retryOptions, facilityId);
+
+    /**
+     * Check 1 - Facility stage `07` only
+     * Ensure facility is `Issued` before processing amendment payload
+     */
+    const { facilityStageCode } = fmr;
+
+    if (!acceptableFacilityStage.includes(facilityStageCode)) {
+      // Error upon unacceptable facility stage
+      console.error('Facility %s stage is %s, amendment will not be processed', facilityId, facilityStageCode);
```

Comment:
I'm leaving this as not using the `incorrectFacilityStageErrorMessage` to follow existing patterns, but am very happy to change this